frankenstein/notes.md

```diff
@@ -0,0 +1,40 @@
+= Project Frankenstein
+
+Design Doc: https://docs.google.com/document/d/1C7yhMnb1x2sfeoe45f4mnnKConvroWhJ8KQZwIHJOuw/edit#heading=h.f7lkb8wswewc
+
+== Retrieval
+
+Use existing prometheus binary; add a --retrieval-only flag to existing prometheus?  and use one of the remote storage protocols or add a new one.
```

Comment:
Determining modes based on flags across the entire flag set will get messy.
Our util/cli tool should provide subcommands in a limited way.

I've been thinking about switching all our tools over to cobra and pflag – should be a drop in replacement and way more powerful for these things.

As with anything, whatever helps you move forward is fine. Just shouldn't be a final solution.